File: sdk-java/hfcx-sdk-client/src/main/java/eg/gov/healthflow/hfcx/sdk/client/Status.java
package eg.gov.healthflow.hfcx.sdk.client;

/**
 * Outcome of an HFCX outbound request as observed at the SDK call site.
 *
 * <p>Mapping to platform behaviour:
 * <ul>
 *   <li>{@link #ACCEPTED} — gateway returned HTTP 202; the recipient will
 *       process asynchronously.</li>
 *   <li>{@link #REJECTED} — gateway or recipient returned a 4xx response
 *       attached to a typed exception; the response object is surfaced
 *       only when the caller chose to catch the exception.</li>
 *   <li>{@link #STUBBED} — used by Sprint J3 stubs that construct headers
 *       and propagate correlation IDs but do not yet POST to the gateway.
 *       Replaced by {@link #ACCEPTED} or {@link #REJECTED} when Sprint J4
 *       lands the outbound encryption + HTTP path.</li>
 * </ul>
 */
public enum Status {
    ACCEPTED,
    REJECTED,
    STUBBED
}

File: sdk-dotnet/src/HealthFlow.Hfcx.Sdk/Client/Status.cs
// Copyright (c) HealthFlow Medical HCX. Licensed under Apache 2.0.

namespace HealthFlow.Hfcx.Sdk.Client;

/// <summary>
/// Outcome of an HFCX outbound request as observed at the SDK call site.
/// Cross-SDK invariant with Java's <c>Status</c> and Python's
/// <c>Status</c> enum.
/// </summary>
public enum Status
{
    /// <summary>Gateway returned HTTP 202 (Accepted).</summary>
    Accepted,

    /// <summary>Gateway returned a 4xx — the recipient or platform rejected the request.</summary>
    Rejected,

    /// <summary>Reserved: stubbed-mode response from a test harness.</summary>
    Stubbed,
}
